common/primitives/src/utils.rs

```diff
@@ -19,6 +19,32 @@ pub mod as_hex {
 	}
 }
 
+/// Handle serializing and deserializing from `Option<Vec<u8>>` to hexadecimal
+#[cfg(feature = "std")]
+pub mod as_hex_option {
+	use serde::{Deserializer, Serializer};
+
+	/// Serializes a `Vec<u8>` into a hexadecimal string
+	pub fn serialize<S>(bytes: &Option<Vec<u8>>, serializer: S) -> Result<S::Ok, S::Error>
+	where
+		S: Serializer,
+	{
+		match bytes {
+			Some(bytes) => impl_serde::serialize::serialize(bytes.as_slice(), serializer),
+			None => serializer.serialize_none(),
+		}
+	}
+
+	/// Deserializes a hexadecimal string into a `Vec<u8>`
+	pub fn deserialize<'de, D: Deserializer<'de>>(
+		deserializer: D,
+	) -> Result<Option<Vec<u8>>, D::Error>
+	where
+		D: Deserializer<'de>,
+	{
+		impl_serde::serialize::deserialize(deserializer).map(|r| Some(r))
```

Comment:
How do we get a None result and an Error?

Reply:
we really don't, because this deserializer isn't called when using `skip_serializing_if = "Option::is_none"`